src/main/java/com/icthh/xm/uaa/web/rest/AccountResource.java

```diff
@@ -271,6 +275,29 @@ public String isAuthenticated(HttpServletRequest request) {
         return new ResponseEntity<>(HttpStatus.OK);
     }
 
+    /**
+     * POST /account/reset_password/init : Reset password and send reset key via customizable transport/flow.
+     *
+     * @param request password reset request
+     * @return the ResponseEntity with status 200 (OK) if the password was reset and reset flow (if configured) is executed
+     */
+    @PostMapping(path = "/account/reset_password/init", produces = MediaType.APPLICATION_JSON_UTF8_VALUE, consumes = MediaType.APPLICATION_JSON_VALUE)
+    @Timed
+    @PreAuthorize("hasPermission({'login': #request.login}, 'ACCOUNT.PASSWORD.RESET')")
+    @PrivilegeDescription("Privilege to reset password and start customizable reset flow")
+    public ResponseEntity<Void> requestPasswordResetViaRequestedFlow(@RequestBody ResetPasswordVM request) {
```

Comment:
Suggest to consider rewrite logic in the more functional way (avoiding mixing imperative and functional styles as much as possible):

```
public ResponseEntity<Void> requestPasswordResetViaRequestedFlow(@RequestBody ResetPasswordVM request) {

        UserLoginType userLoginType = UserLoginType.valueOfType(request.getLoginType())
                                                   .orElseThrow(() -> new BusinessException(INCORRECT_LOGIN_TYPE));

        userService.requestPasswordResetForLoginWithType(request.getLogin(), userLoginType)
                   .map(user -> new PasswordResetRequest(request.getResetType(), user))
                   .ifPresent(userService::handlePasswordReset);

        return new ResponseEntity<>(HttpStatus.OK);
    }
```

Reply:
done